### DISCUSSION TITLE
failureCount value reset

### DISCUSSION THREAD

**Discussion by OmerCohenAviv** [TanStack/query#3525] [Q&A] [answered]
Hey!

I have noticed that "failureCount" is set to 0 after refetchInterval is invoked, is this intended?
If not it's not mentioned in the DOCS, I have  made a pull-request but closed it for now until it's more clear

**Comment by TkDodo** [comment, marked as answer]
`failureCount` is the "internal" counter that tells you how often a query has failed while doing retries. For example, when if you have 3 retries configured (the default), the counter will go from 0 to 4. if it's on e.g. `2`, your query will still be in `loading` or `fetching` state, but you can use this info to show a "something is taking longer than expected...` or so message to the user, if you want to.

the `failureCount` is also passed to the `retry` function so that you know how often it has failed.

because of that, when a new fetch starts for whatever reason (can be a call to `refetch`, can be manual invalidation, can be a background refetch on window focus or, like in your case, an interval fetching), the count goes back to `0` because for this fetch, no failures have occurred.

the docs are correct in saying:

```
failureCount: number
  The failure count for the query.
  Incremented every time the query fails.
  Reset to 0 when the query succeeds.
```

because that's what's happening :)

**Reply by TkDodo** [reply]
the query state has an internal `errorUpdateCount` that we do not expose for some reason on `useQuery`. would that help?

**Reply by Omercoav** [reply]
> the query state has an internal `errorUpdateCount` that we do not expose for some reason on `useQuery`. would that help?

sorry deleted my last comment by mistake,

_"Thanks a lot for the detailed answer @TkDodo ,
another question if I may,
Do we have an ability to get the total failureCount we have - sum of all internals queries?
it could help on case such - A widget that has a refetch interval and shows a "unresponsive" widget UI after X amount of fails."_



Yes it would help, there won't be a need to manage another state if we expose `errorUpdateCount`

**Reply by TkDodo** [reply]
if you want, you can open a PR for that. It should be as simple as returning `errorUpdateCount` here:

https://github.com/tannerlinsley/react-query/blob/4d753c01c3fa3bb0fb7b6e8935e2a656763b2811/src/core/queryObserver.ts#L548

which already exists on the internal `state`. Of course, some docs update and tests would be nice :)

**Reply by Omercoav** [reply]
on it, thanks!
